src/utils/helpers/evaluateElements.ts

```diff
@@ -0,0 +1,89 @@
+import evaluateExpression, { isEvaluationExpression } from '@openmsupply/expression-evaluator'
+import config from '../../config.json'
+import {
+  EvaluatedElement,
+  ResponsesByCode,
+  User,
+  ApplicationDetails,
+  ElementForEvaluation,
+  EvaluationOptions,
+  EvaluatorNode,
+} from '../types'
+const graphQLEndpoint = config.serverGraphQL
+
+type PartialEvaluatedElement = Partial<EvaluatedElement>
+type EvaluationObject = {
+  responseObject?: ResponsesByCode
+  currentUser?: User | null
+  applicationData?: ApplicationDetails
+}
+
+type EvaluateElements = (
+  elements: ElementForEvaluation[],
+  evaluationOptions: EvaluationOptions,
+  objects: EvaluationObject
+) => Promise<PartialEvaluatedElement[]>
+
+const evaluationMapping: { [resultKey in keyof EvaluatedElement]: keyof ElementForEvaluation } = {
+  isEditable: 'isEditableExpression',
+  isRequired: 'isRequiredExpression',
+  isVisible: 'isVisibleExpression',
+  isValid: 'validationExpression',
+  defaultValue: 'defaultValueExpression',
+}
+
+export const evaluateElements: EvaluateElements = async (elements, evaluationOptions, objects) => {
+  const elementPromiseArray: Promise<PartialEvaluatedElement>[] = []
+  elements.forEach((element) => {
+    elementPromiseArray.push(evaluateSingleElement(element, evaluationOptions, objects))
+  })
+  return await Promise.all<PartialEvaluatedElement>(elementPromiseArray)
+}
+
+type EvaluatElement = (
+  element: ElementForEvaluation,
+  evaluationOptions: EvaluationOptions,
+  objects: EvaluationObject
+) => Promise<PartialEvaluatedElement>
+
+const evaluateSingleElement: EvaluatElement = async (
+  element,
+  evaluationOptions,
+  { responseObject, currentUser, applicationData }
```

Comment:
Any particular reason for the name "responseObject" as opposed to "allResponses" or "responsesByCode". At the very least could we capture the fact that it contains _multiple_ objects by pluralising it -- `responsesObject` perhaps?

Reply:
sure, it was like this originally so didn't change, good time to change now.
d8f386dfa020174179752f735f2226fb1da7f0f3